idaes/generic_models/properties/core/reactions/equilibrium_constant.py

```diff
@@ -141,10 +142,11 @@ def build_parameters(rblock, config):
                 "is only supported for molarity or activity forms. "
                 "Currently selected form: {}".format(rblock.name, c_form))
 
-        rblock.dh_rxn_ref = Var(
-                doc="Specific molar heat of reaction at reference state",
-                units=units["energy_mole"])
-        set_param_from_config(rblock, param="dh_rxn_ref", config=config)
+        # Check that heat of reaction is constant
+        if config.heat_of_reaction is not constant_dh_rxn:
+            raise ConfigurationError(
+                "{} calculating equilibrium constants from Gibbs energy "
+                "assumes constant heat of reaction.".format(rblock.name))
```

Comment:
This error message reads more like a warning. I think it should be adjusted to 

"{} calculating equilibrium constants from Gibbs energy requires the user to specify a constant heat of reaction"
 or
"{} calculating equilibrium constants from Gibbs energy assumes constant heat of reaction, but the user has not specified that the heat of reaction should be constant"